docs/spec/spec.md

```diff
@@ -0,0 +1,251 @@
+
+The conforming implementation of the specification is included in the Ballerina distribution. Any deviations from the specification are considered bugs.
+
+## Contents
+
+1. [Introduction](#1-introduction)
+2. [Components](#2-components)
+   - 2.1 [RFC Client](#21-rfc-client)
+     - 2.1.1 [Configurations](#211-configurations)
+     - 2.1.2 [Available Operations](#212-available-operations)
+       - 2.1.2.1 [Execute a Function Module via RFC](#2121-execute-a-function-module-via-rfc)
+       - 2.1.2.2 [Send IDocs to an SAP System](#2122-send-idocs-to-an-sap-system)
+     - 2.1.3 [Data Types](#213-data-types)
+   - 2.2 [Listener](#22-listener)
+     - 2.2.1 [Configurations](#221-configurations)
+     - 2.2.2 [Available Operations](#222-available-operations)
+       - 2.2.2.1 [Receive IDocs](#2221-receive-idocs)
+       - 2.2.2.2 [Error Handling in Listener](#2222-error-handling-in-listener)
+
+## 1. Introduction
+
+The Ballerina SAP JCo Connector provides seamless integration with SAP systems through Java Connector (JCo) capabilities. This connector allows Ballerina applications to interact with SAP systems, enabling operations like Remote Function Call (RFC) execution, IDoc processing, and more. By leveraging this connector, developers can easily integrate SAP functionalities into their Ballerina applications, making it an essential tool for enterprises working with SAP.
+
+The `ballerinax/sap.jco` package exposes the SAP JCo library as Ballerina functions, enabling easy access to SAP's enterprise resource planning (ERP) software and other SAP solutions, such as human capital management (HCM), customer relationship management (CRM), and supply chain management (SCM).
+
+### Architecture Overview
+
+The SAP JCo Connector architecture consists of key components that facilitate communication between Ballerina applications and SAP systems. These components include the SAP function modules, the SAP system itself, Ballerina code that handles the RFC calls, and the IDoc servers responsible for sending and receiving IDocs. This architecture enables robust integration, supporting various enterprise operations.
+
+## 2. Components
+
+### 2.1 RFC Client
+
+#### 2.1.1 Configurations
+
+##### 2.1.1.1 DestinationConfig
+
+The `DestinationConfig` type holds the configuration details needed to create a BAPI connection.
+
+- **ashost**: The SAP host name (`jco.client.ashost`).
+- **sysnr**: The SAP system number (`jco.client.sysnr`).
+- **jcoClient**: The SAP client (`jco.client.client`).
+- **user**: The SAP user name (`jco.client.user`).
+- **passwd**: The SAP password (`jco.client.passwd`).
+- **lang**: The SAP language (`jco.client.lang`). Default is "EN".
+- **group**: The SAP group (`jco.client.group`). Default is "PUBLIC".
```

Comment:
Rather doing this shall we add the relevant Ballerina record with proper documentation ?

Reply:
Yes. That would be better, Changed with [1a3f2d5](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/56/commits/1a3f2d588a8e26fea96920ea4152397a479440e6).